### ISSUE TITLE
Unbound with ocserv

### ISSUE THREAD

**Issue by williclarkam** [fanyueciyuan/eazy-for-ss#10]
Hello again Sir, 
I wanted to know, is there any way to use unbound DNS with ocserv? My unbound server runs for example on 172.16.30.0/24 now when I add this ip no domain will be solved, but this work great on openvpn. 

Any help would be greatly appreciated. 
Thanks a lot.

**Comment by liyangyijie** [comment]
Hi,
Edit the /etc/ocserv/ocserv.conf file,please.
```
# Whether to tunnel all DNS queries via the VPN. This is the default 
# when a default route is set. 
#tunnel-all-dns = true

# The advertized DNS server. Use multiple lines for 
# multiple servers. 
# dns = fc00::4be0 
dns = 172.16.30.0
```
Thanks for your issue.

**Comment by williclarkam** [comment]
Thanks for the reply.
Unfortunatly this doesn't work. I have pinged an IP address and a Domain name, the domain name would NOT be resolved but the IP have access, so I don't think this issue is related to iptables but the DNS server!

```
C:\Users\myUsername>ping google.com
Ping request could not find host google.com. Please check the name and try again.

```
As you can see pinging 8.8.8.8 works perfectly!

```
C:\Users\myUsername>ping 8.8.8.8

Pinging 8.8.8.8 with 32 bytes of data:
Reply from 8.8.8.8: bytes=32 time=206ms TTL=56
Reply from 8.8.8.8: bytes=32 time=204ms TTL=56
Reply from 8.8.8.8: bytes=32 time=206ms TTL=56
Reply from 8.8.8.8: bytes=32 time=628ms TTL=56

Ping statistics for 8.8.8.8:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 204ms, Maximum = 628ms, Average = 311ms
```
* Oh, almost forgot, I did change the ipv4 pool too.
the default was 192.168.10.0/24
I have changed it to my DNS server pool, though this may helps, solving the issue, but no luck :(

Best Regards,

**Comment by liyangyijie** [comment]
Hi,

If the DNS server and the ocserv are in the same LAN， [it](https://github.com/fanyueciyuan/eazy-for-ss/blob/master/ocservauto/ocserv-up.sh#L18) will disable LAN  communication.
Please change the ocserv-up.sh (Line 18) as below
```
iptables -t nat -A POSTROUTING -s $ocserv_ip4_work_mask -m comment --comment "$gw_intf_oc (ocserv)" -j MASQUERADE
```
Just del ```! -d $ocserv_ip4_work_mask```



Then,reload iptables or restart ocserv.

Thanks for your feedback.

**Comment by williclarkam** [comment]
Thanks Sir, but still no ping, is it possible, that there is some issue with unbound config?

**Comment by williclarkam** [comment]
UPDATE: I think I have found the problem, but not the solution :D
I tried to add interface: 192.168.10.0 to unbound default config , but then unbound did NOT run at all, then I ran the command `ifconfig `and there was no interface with the IP 192.168.10.0 and subnet 255.255.255.0 so it's interesting how does even the traffic go out, from which interface?!
e.g. OpenVPN creates tun0 interface with the default IP of 10.8.0.1/24 and the traffic goes out from this interface.
But I am still new to all this, so excuse me, if I am completely wrong.
Best Regards,
